### PR TITLE
chore(website): remove PostHog in favor of Mixpanel

### PR DESCRIPTION
**Pull request by jamilbk**
* Remove PostHog from product
* Remove PostHog from website
* Add Mixpanel to website

Why? PostHog is a bit too much overhead for simple analytics for us for now, and some of the bugs we hit prevented us from using certain workflows.

We are still tracking `ping` events from legacy instances in PostHog.

### CODE REVIEW

File: elixir/apps/domain/mix.exs [hunk 1]
```diff
@@ -78,6 +75,7 @@ defmodule Domain.MixProject do
       # https://github.com/elixir-lang/elixir/issues/12777
       # Remove Swoosh from Domain once this is fixed.
       {:phoenix_swoosh, "~> 1.0"},
+      {:hackney, "~> 1.9"},
```
Comment (jamilbk) [inline elixir/apps/domain/mix.exs:78]: Needed to fix https://github.com/firezone/firezone/actions/runs/7348790224/job/20007576526#step:18:16

File: elixir/mix.lock
Comment (jamilbk) [inline elixir/mix.lock:1]: I ended up doing a cache bust by just updating dependencies to fix the weird errors I was seeing here:

https://github.com/firezone/firezone/actions/runs/7348750997/job/20007446986#step:9:12

File: website/src/components/Providers/index.jsx [hunk 2]
```diff
@@ -1,37 +1,16 @@
...
-      let url = window.origin + pathname;
-      if (searchParams.toString()) {
-        url = url + `?${searchParams.toString()}`;
-      }
-      posthog.capture("$pageview", {
-        $current_url: url,
-      });
-    }
-  }, [pathname, searchParams]);
+  const token = "b0ab1d66424a27555ed45a27a4fd0cd2";
```
Comment (bmanifold) [inline website/src/components/Providers/index.jsx:6]: I'm guessing this token isn't sensitive, but just wanted to make sure.

**Review by bmanifold — COMMENTED** [review]
Overall, LGTM, just had one small question.

File: website/src/components/Providers/index.jsx (hunk 2 above)
Comment (jamilbk) [reply website/src/components/Providers/index.jsx:6]: Correct, it's just a public write token

**Review by bmanifold — APPROVED** [review]
LGTM

**Pull request merged**